### PR TITLE
Add support for a 3rd argument `compression` at the `percentile()` aggregation

### PR DESCRIPTION
**Pull request by seut**
This allows a user to control the accuracy of the result by trading off slightly higher memory usages.

Relates to #17137.

### CODE REVIEW

**Comment by seut** [comment]
Moved this from a feature to a fix by https://github.com/crate/crate/pull/17156/commits/4968888588e7c315fac94992b62148337c4ea78d.

**Review by henrikingo — APPROVED** [review]

File: server/src/main/java/io/crate/execution/engine/aggregation/impl/PercentileAggregation.java [hunk 1]
```diff
@@ -107,6 +137,10 @@ public TDigestState iterate(RamAccounting ramAccounting,
                                 Input<?>... args) throws CircuitBreakingException {
         if (state.isEmpty()) {
             Object fractionValue = args[1].value();
+            if (args.length > 2) {
```
Comment (mkleen) [inline server/src/main/java/io/crate/execution/engine/aggregation/impl/PercentileAggregation.java:140]:
```suggestion
            if (args.length == 3) {
```

Minor: This is more precise.

File: docs/appendices/release-notes/5.10.0.rst [hunk 2]
```diff
@@ -108,7 +108,7 @@ Data Types
 Scalar and Aggregation Functions
 --------------------------------
 
-None
+- None
```
Comment (mkleen) [inline docs/appendices/release-notes/5.10.0.rst:111]: This can be reverted.

File: docs/appendices/release-notes/5.9.6.rst [hunk 3]
```diff
@@ -52,3 +52,12 @@ Fixes
 
 - Fixed a possible deadlock where concurrent delete requests could take a lock but then never
   release it, leading to requests timing out and threads parked in infinite waits.
+
+- Fixed a regression introduced with :ref:`version_5.6.0` which caused the
```
Comment (mkleen) [inline docs/appendices/release-notes/5.9.6.rst:56]: The regression was caused because of a new default in the library right?

Comment (seut) [reply docs/appendices/release-notes/5.9.6.rst:56]: It was caused by a new version of the library, see https://github.com/crate/crate/issues/17137#issuecomment-2548861318.

Comment (mkleen) [reply docs/appendices/release-notes/5.9.6.rst:56]: We could mention this in the note.

File: server/src/main/java/io/crate/execution/engine/aggregation/impl/PercentileAggregation.java (hunk 1 above)
Comment (seut) [reply server/src/main/java/io/crate/execution/engine/aggregation/impl/PercentileAggregation.java:140]: But would need adjusted if one would add another argument. I'd leave it like that.

File: docs/appendices/release-notes/5.9.6.rst (hunk 3 above)
Comment (seut) [reply docs/appendices/release-notes/5.9.6.rst:56]: How will this help?

Comment (mkleen) [reply docs/appendices/release-notes/5.9.6.rst:56]: It would just point out the root cause. it's ok like this for me.

**Pull request merged**